examples/vault-cluster-private/main.tf

```diff
@@ -123,9 +123,10 @@ data "template_file" "user_data_consul" {
 # ---------------------------------------------------------------------------------------------------------------------
 
 data "aws_vpc" "default" {
-  default = true
+  default = "${var.vpc_id == "" ? true : false}"
+  id = "${var.vpc_id}"
```

Comment:
Hm, I'm worried what happens if `id` is set to an empty string. Does Terraform ignore it, or does it complain? Unfortunately, Terraform is [inconsistent with how it treats empty vs omitted params](https://github.com/hashicorp/terraform/issues/14037), so I'm not sure if this will cause problems for people trying to use the default VPC. Did you try it that way?

Reply:
No, I didn't try it that way - good catch. I'd have to create a default VPC, which I'm not sure is a reversible process in a single AWS account.

Reply:
http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/default-vpc.html#create-default-vpc

Reply:
Hmm. I'm not comfortable changing my present employer's setup to create and delete a default VPC, since it seems to indicate a bunch of other things are touched by that process (subnets, security groups, etc.), and I don't know the implications of that (or the effect on other Terraform users). Is there another way to test this, perhaps with a free-tier?

Reply:
Sure, create a brand new AWS account with the free tier and it'll have a default VPC built-in.